### PR TITLE
Check actual stop/platform elements role in routes

### PR DESCRIPTION
**Pull request by alexey-zakharenkov**
Resolves issue #85.

Having been run on all networks, the updated code revealed another 2 missed (not connected to a station) stops which were silently ignored before. Also stops/platforms with the following incorrect roles were warned: 'start', 'station', 'access', 'Bayrampaşa - Maltepe'; 'stop' for a platform and 'platform' for a stop_position. Also rails with public_transport=stop_position were detected.

### CODE REVIEW

File: subway_structure.py [hunk 1]
```diff
@@ -664,16 +664,18 @@ def __init__(self, relation, city, master=None):
...
-                el_type = RouteStop.get_member_type(el, m['role'], city.modes)
-                if el_type:
+            el = city.elements[k]
+            actual_role = RouteStop.get_actual_role(el, m['role'], city.modes)
+            if actual_role:
+                if m['role'] and actual_role not in m['role']:
+                    city.warn("Wrong role '{}' for {} {}".format(m['role'], actual_role, k), relation)
+                if k in city.stations:
+                    st_list = city.stations[k]
+                    st = st_list[0]
```
Comment (syershov) [inline subway_structure.py]: Тут может вылететь исключение при пустом списке

Comment (alexey-zakharenkov) [reply subway_structure.py]: In fact, it cannot. The `st_list` aka `city.stations[k]` list, being a value in a defaultdict, is created only in this line of code:
`self.stations[st_el].append(station)`
at the moment of the first insertion into it, and no deletions are made afterwards.

**Comment by alexey-zakharenkov** [comment]
I've reverted many changes as they turned out incorrect, preserving only one considerable change that resolves namely the issue: replacement of
`if 'stop' in m['role'] or 'platform' in m['role']:`
condition with
`if actual_role:`
in line 738 of the changed file.

File: subway_structure.py [hunk 2]
```diff
@@ -735,16 +737,17 @@ def __init__(self, relation, city, master=None):
             if 'tags' not in el:
                 city.error('Untagged object in a route', relation)
                 continue
-            if 'stop' in m['role'] or 'platform' in m['role']:
+            actual_role = RouteStop.get_actual_role(el, m['role'], city.modes)
+            if actual_role:
```
Comment (Zverik) [inline subway_structure.py]: `if actual_role in ('stop', 'platform')`?

Comment (alexey-zakharenkov) [reply subway_structure.py]: `actual_role` may take values from ` {'stop', 'platform', None}` so I can't find any difference. Do you see some logic or better maintainability in explicit specifying of possible values?

Also, I don't like duplicating of these two lines in one function:
*) `el = city.elements[k]`
*) `actual_role = RouteStop.get_actual_role(el, m['role'], city.modes)`
and wanna eliminate it in this PR by rearranging some blocks of code, so please wait approving thus far.

Comment (Zverik) [reply subway_structure.py]: Okay, I thought there was an option of "station", but now I see there isn't.

**Comment by alexey-zakharenkov** [comment]
The things occurred no that easy. Simple replacement of a member role check with "actual role" notion has helped to detect some _"Feature is not connected to a station in route"_ objects but shadows two other problems:
* _"An under construction {} in route"_ is not longer reported since a feature under construction has None actual role.
* _"Missing station={mode} on a station"_ is not longer reported since such a station fails on is_station() check and hence has None actual role.

I propose to separate these 3 diagnostic cases, and always to ignore declared member role.

Also I offer to allow under construction (actually - any) objects with 'inactive' role in routes. At least one OSMer opposes removing of stops under construction from routes and adds them with 'stop_inactive' role, another one follows him.

Also note new _"Unauthorized member in route"_ diagnostic message. It already revealed such members in route relations as: parkings, train_station buildings, stop_area relations, another route relations, highways, shops, and many rails in construction/proposed status. At this moment the validation page demonstrates the scale of the disaster (about 50 networks got broken).

File: subway_structure.py [hunk 3]
```diff
@@ -735,16 +739,27 @@ def __init__(self, relation, city, master=None):
...
+            if el['tags'].get('railway') in ('station', 'halt'):
+                city.error('Missing station={} on a {}'.format(self.mode, m['role']), el)
+            else:
+                actual_role = RouteStop.get_actual_role(el, m['role'], city.modes)
+                if actual_role:
                     city.error('{} {} {} is not connected to a station in route'.format(
-                        m['role'], m['type'], m['ref']), relation)
+                        actual_role, m['type'], m['ref']), relation)
+                elif not StopArea.is_track(el):
+                    city.error('Unauthorized member {} {} in route'.format(m['type'], m['ref']), relation)
```
Comment (Zverik) [inline subway_structure.py:762]: This message is unclear. Maybe "Unknown member type ..."?

**Pull request merged**